SimTKcommon/tests/TestStateXml.cpp

```diff
@@ -25,6 +25,46 @@
 
 using namespace SimTK;
 
+/* This function converts the provided instance of `T` to Xml, deserializes
+it from Xml, ensures the deserialized instance is equal to the original
+instance, reserializes the deserialized instance back to Xml and ensures
+the original desrialization and the reserialization are equal.
+
+One might set `checkInMemoryEquality` to false if `value` contains NaNs. */
```

Comment:
I have created issue #505 for that. 

I realized that `if (x.isNaN())` is not a good way to remove the need for that extra bool argument because `x` may not have an `isNaN()`. And yes, for now, `NaN`s are checked via the string comparison (it's actually why I added the string comparison).

Let me know if you think there is still something to change about this method for this PR.